src/apps/finance/hooks/useBalances.js

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useMemo, useState } from 'react'
+import { useConnect } from '@1hive/connect-react'
+import { useOrganizationState } from '@/providers/OrganizationProvider'
+import { useContractReadOnly, getContract } from '@/hooks/shared/useContract'
+import { getAppByName } from '@/utils/app-utils'
+import BN from 'bn.js'
+import usePromise from '@/hooks/shared/usePromise'
+// import { toMs } from '@/utils/date-utils'
+
+import vaultBalanceAbi from '../abi/vault-balance.json'
+import minimeTokenAbi from '@/abi/minimeToken.json'
+import { useMounted } from '@/hooks/shared/useMounted'
+
+const VAULT_NAME = 'agent'
+const INITIAL_TIMER = 2000
+
+const CHAIN_TOKEN_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+const useBalances = (timeout = 7000) => {
+  const { apps, connectedFinanceApp } = useOrganizationState()
+  const [tokenBalances = [], { loading, error }] = useConnect(
+    () => connectedFinanceApp?.onBalance(),
+    [connectedFinanceApp]
+  )
+  const mounted = useMounted()
+  const [tokenData, setTokenData] = useState([])
+  const [tokenWithBalance, setTokenWithBalance] = useState([])
+  const [loadingBalances, setLoadingBalances] = useState(true)
+
+  const chainId = 100 // TODO HANDLE CHAINS
+  const vaultContract = useContractReadOnly(
+    getAppByName(apps, VAULT_NAME).address,
+    vaultBalanceAbi,
+    chainId
+  )
+
+  useEffect(() => {
+    if (!tokenBalances?.length || !mounted) {
```

Comment:
Let's set the main loading to false when the connector returns no token balance.
```suggestion
    if (!tokenBalances?.length || !mounted) {
      if (!tokenBalances?.length && !loading) {
        setLoadingBalances(false)      
      }
```

Reply:
good catch!

Reply:
ended up removing that balance length condition, since is not necessary if we set the loading balances there in false it generates a loop since it does not give the time for the previous effect to end.